src/Raven.Server/Dashboard/DatabasesInfo.cs

```diff
@@ -21,6 +22,25 @@ public override DynamicJsonValue ToJson()
             json[nameof(Items)] = new DynamicJsonArray(Items.Select(x => x.ToJson()));
             return json;
         }
+
+        public override DynamicJsonValue ToJsonWithFilter(Func<string, bool> filter)
```

Comment:
Why not return an empty array?

Reply:
if we return null, then we don't append anything to the queue

Reply:
moreover we need to filter inner structures and we don't want to send database names users can't access.